### ISSUE TITLE
Add headless browser tests

### ISSUE THREAD

**Issue by WesCossick** [sparksuite/react-accessible-dropdown-menu-hook#23]
Some of the component's behavior relies on how the browser behaves. Enzyme cannot simulate this, so we'll rely on headless browser tests for that type of testing. Before we can implement this, though, we'll need to create a demo website (#9).

**Comment by corymharper** [comment]
It appears puppeteer is not correctly building the react app. I have tried a variety of methods but here is an example that shows the content is never loading

```js
it('should contain a button with an id of menu', async () => {
	await page.waitFor("#menu");
});
```

This test hangs with the error:
`Timeout - Async callback was not invoked within the 5000ms timeout specified by jest.setTimeout.Timeout - Async callback was not invoked within the 5000ms timeout specified by jest.setTimeout.Error`

This could be an issue of the timeout being too short, but it seems more likely that content is just never reaching the page. I will investigate further by increasing the timeout.

**Comment by WesCossick** [comment]
You're not calling `expect`. See: https://jestjs.io/docs/en/tutorial-async#asyncawait

**Comment by corymharper** [comment]
I don't think that is the issue in this instance, by moving the same line into the `beforeAll` I get the same error.

```js
// Tests
beforeAll(async () => {
	await page.goto(`file://${path.join(__dirname, '..', '..', 'docs', 'build', 'index.html')}`);
	await page.waitFor("#menu");
});
```

via the puppeteer docs, this [method](https://pptr.dev/#?product=Puppeteer&version=v2.0.0&show=api-pagewaitforselectororfunctionortimeout-options-args) should just wait for that selector to appear on the page. I have been looking at [this](https://github.com/smooth-code/jest-puppeteer/blob/master/examples/create-react-app/jest-puppeteer.config.js) example to try to figure something out, but trying to sort out some errors.

**Comment by WesCossick** [comment]
Can you have Puppeteer print out the HTML source of what it's loading after it executes all the page scripts?

**Comment by corymharper** [comment]
```html
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<link rel="icon" href="/favicon.ico">
<meta name="viewport" content="width=device-width,initial-scale=1">
<title>React Accessible Dropdown Menu Hook</title>
<link href="/static/css/main.be4a93f6.chunk.css" rel="stylesheet">
</head>

<body>
<noscript>You need to enable JavaScript to run this app.</noscript>
<div id="root"></div>
<script>!function(i){function e(e){for(var r,t,n
=e[0],o=e[1],u=e[2],l=0,f=[];l<n.length;l++)t=n[l],Object.prototype.hasOwnProperty.call(p,t)&&p[t]&&f.push(p[t][0]),p[t]=0;for(r in o)Object.prototype.hasOwnProperty.call(o,r)&&(i[r]=o[r]);for(s&&s(e);f.length;)f.shif
t()();return c.push.apply(c,u||[]),a()}function a(){for(var e,r=0;r<c.length;r++){for(var t=c[r],n=!0,o=1;o<t.length;o++){var u=t[o];0!==p[u]&&(n=!1)}n&&(c.splice(r--,1),e=l(l.s=t[0]))}return e}var t={},p={1:0},c=[];f
unction l(e){if(t[e])return t[e].exports;var r=t[e]={i:e,l:!1,exports:{}};return i[e].call(r.exports,r,r.exports,l),r.l=!0,r.exports}l.m=i,l.c=t,l.d=function(e,r,t){l.o(e,r)||Object.defineProperty(e,r,{enumerable:!0,g
et:t})},l.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},l.t=function(r,e){if(1&e&&(r=l(r)),8&e)return r;if(4&e&&"object"==typeof r&&r&&r.__esModule)return r;var t=Object.create(null);if(l.r(t),Object.defineProperty(t,"default",{enumerable:!0,value:r}),2&e&&"string"!=typeof r)for(var n in r)l.d(t,n,function(e){return r[e]}.bind(null,n));return t},l.n=function(e){var r=e&&e.__esModule?function(){return e.default}:function(){return e};return l.d(r,"a",r),r},l.o=function(e,r){return Object.prototype.hasOwnProperty.call(e,r)},l.p="/";var r=this.webpackJsonpdemo=this.webpackJsonpdemo||[],n=r.push.bind(r);r.push=e,r=r.slice();for(var o=0;o<r.length;o++)e(r[o]);var s=n;a()}([])</script><script src="/static/js/2.f8ed4020.chunk.js"></script>
<script src="/static/js/main.41dcf1df.chunk.js"></script>
</body>
</html>
```

I tried to separate it out a bit, it came in as a huge block of text.

**Comment by corymharper** [comment]
Another interesting note is that waiting for the "load" event from the page also times out.

**Comment by WesCossick** [comment]
I think this is the issue:

```
<script src="/static/js/main.41dcf1df.chunk.js"></script>
```

That's not a relative path, so it won't be able to load `/static/`.

**Comment by corymharper** [comment]
Possible solutions to that problem? That path is automatically generated when the app is built, can the path be specified?

**Comment by corymharper** [comment]
My current working solution is to add a jest-puppeteer config that runs it's own dev server. I will push up that code now so you can take a look.

**Comment by WesCossick** [comment]
I'm not a big fan of that solution, because it's a bit more complicated and it runs against the development version of the app instead of the fully-built version.

A quick Google search turned up some results that may allow us to tell CRA to use relative paths when building. I'm going to test some of those suggestions out.

**Comment by WesCossick** [comment]
Okay, that bug has been fixed in #32.
